### PR TITLE
Unhardcode CompositionProvider (fix #79)

### LINKED ISSUE #79: Prepare for working with BlendingProvider
Starting from this week, the composition information can be delivered in one of the following JSON keys:

```txt
CompositionProvider
CompositionProvider_0
BlendingProvider_0
BlendingProvider_1
```

Please prepare `starcounter-include` for searching for the composition information in all of the above combinations. The key will not change during the session, so if you use searching, it can be searched once.

This is a requirement of https://github.com/Starcounter/RebelsLounge/issues/232.

### PR DESCRIPTION
**Pull request by tomalec**
It could be configured per instance as well as globally by updating the prototype.

remaining: update tests, add new one

Related issues:
https://github.com/Starcounter/starcounter-include/issues/30
https://github.com/Starcounter/starcounter-include/issues/43
https://github.com/Starcounter/starcounter-include/issues/79

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,16 +135,17 @@ Attribute     | Options  | Default      | Description
 Property   | Options           | Default | Description
 ---         | ---               | ---     | ---
 `partial`   | *Object*          |         | Object containing partial view-model, bindable with Polymer
-`partialId` | *String*          |         | Partial Id used to identify partial, usually it's fetched from `partial.CompositionProvider.PartialId`.
+`partialId` | *String*          |         | Partial Id used to identify partial, usually it's fetched from `partial.{CompositionProvider}.PartialId`.
 `viewModel` | *Object*          |         | Alias for `partial`
+`CompositionProvider` | *String* | `CompositionProvider_0` | Key/app name of composition provider. could be overwritten per instance `scInclude.CompositionProvider` or globally by changing the prototype, like: `customElements.get('starcounter-include').prototype.CompositionProvider = 'CustomProvider_7'`
```
Comment (warpech) [inline README.md]: I think this property name should be lowercase, as all other property names.

**Comment by warpech** [comment]
The only way of providing a new value for the namespace name that we have right now, requires overwriting `/sys/starcounter.html` through middleware in BlendingProvider. 

It was identified as an antipattern in https://github.com/Starcounter/CompositionEditor/issues/8 and we want to stop using it.

Do you have some ideas for a better pattern or will we have to use the antipattern for now?

**Review by warpech — CHANGES_REQUESTED** [review]
It will be tough change because it moves the shitty responsibility to determine the namespace to an even more obscure place. But I agree it will make the codebase cleaner.

**Comment by tomalec** [comment]
I would say doing single line 
```js
customElements.get('starcounter-include').prototype.compositionProvider
```
 anywhere in your code is waay less obscure, than 

0. Stop all your apps,
1. Fork `sc-include`,
2. Change hardcoded value,
3. Fork level1,
4. Install forked `sc-include`,
5. Build level1,
6. Start all your apps,
7. Repeat = maintain :finnadie:

**Comment by tomalec** [comment]
> Do you have some ideas for a better pattern or will we have to use the antipattern for now?

Stop pretending we do not have starcounter-specific node in the view-model tree and provide such metadata in view model https://github.com/Starcounter/RebelsLounge/issues/97 possibly in the same convention as Palindrom  metadata

**Comment by warpech** [comment]
> Stop pretending we do not have starcounter-specific node in the view-model tree and provide such metadata in view model Starcounter/RebelsLounge#97 possibly in the same convention as Palindrom metadata

Same thing came to my mind today. We should get back to this. For now, this solution is OK, just please make the property name lowercase.

**Comment by tomalec** [comment]
@warpech  Please check now.

**Review by warpech — APPROVED** [review]
Lgtm

**Comment by tomalec** [comment]
Released in https://github.com/Starcounter/starcounter-include/releases/tag/4.2.0

**Comment by tomalec** [comment]
Tests could be added as part of https://github.com/Starcounter/starcounter-include/issues/43

**Pull request merged**